### PR TITLE
#228 Re-factor the latest filters

### PR DESCRIPTION
**Pull request by alexstark4**
* Shortened class names significantly
* split AsymmetricPeakShapeFilter into two filters
* Shortened class names/descriptions
* provided more options for Treatment
* Shortened class names/descriptions for Criterion

Signed-off-by: Alexander Stark <alexander.stark@lablicate.com>

### CODE REVIEW

**Comment by alexstark4** [comment]
TODO: add Preference Page(s)

@eselmeister @laeubi What is the most efficient method to implement Preference Page(s)? Alex K. suggested using Json to annotate the settings, but that doesn't create Preference Page(s).

**Comment by alexstark4** [comment]
@kerner1000

File: chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/PercentageAreaFilter.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		if(configuration == null) {
+			configuration = createConfiguration(read);
+		}
+		SubMonitor subMonitor = SubMonitor.convert(monitor, read.size());
+		double areaSum = calculateAreaSum(read);
+
+		for(X peak : read) {
+			double compareAreaValue = calculatePercentageAreaCompareValue(peak, areaSum);
+			applySelectedOptions(compareAreaValue, configuration, listener, peak);
+			checkForKeepOption(listener, configuration, peak);
```
Comment (kerner1000) [inline chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/PercentageAreaFilter.java]: Check the keep option inside the "processPeak" method.

File: chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/PercentageAreaFilter.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+		switch (localSettings.getFilterTreatmentOption()) {
+		case ENABLE_PEAK:
+			peak.setActiveForAnalysis(true);
+			listener.updated(peak);
+			break;
+		case DEACTIVATE_PEAK:
+			peak.setActiveForAnalysis(false);
+			listener.updated(peak);
+			break;
+		case KEEP_PEAK:
```
Comment (kerner1000) [inline chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/PercentageAreaFilter.java]: I dont understand this. Why do you modify here the active for analysis property?
As far as I can tell, it should be:
if peak matches filter criterion, do nothing. Else, apply "Delete peak" option.

**Review by kerner1000 — CHANGES_REQUESTED** [review]

**Comment by kerner1000** [comment]
Next time please do changes such as refactoring of names in a dedicated commit.

File: chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/PercentageAreaFilter.java (hunk 2 above)
Comment (alexstark4) [reply chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/PercentageAreaFilter.java]: The idea behind this is that the peak to be kept should be differentiated from the peak to be deleted. Therefore, the property is first modified and then the marked peaks are removed. In this way, the peaks sought are retained.

If I think about it again, this approach is not entirely optimal.

Comment (alexstark4) [reply chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/PercentageAreaFilter.java]: According to Philip, the requirement is that the peaks that are not kept should also be deleted.

File: chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/IntegratedAreaFilter.java [hunk 3]
```diff
@@ -63,34 +63,60 @@ public boolean acceptsIPeaks(Collection<? extends IPeak> items) {
...
 			subMonitor.worked(1);
 		}
 	}
 
 	private static <X extends IPeak> void applySelectedOptions(double peakValue, IntegratedAreaFilterSettings configuration, CRUDListener<X, IPeakModel> listener, X peak) {
 
+		boolean keepFlag = false;
+		if(configuration.getFilterTreatmentOption()==ValueFilterTreatmentOption.KEEP_PEAK) {
+			keepFlag = true;
+		}
```
Comment (alexstark4) [inline chemclipse/plugins/org.eclipse.chemclipse.xxd.model/src/org/eclipse/chemclipse/xxd/model/filter/peaks/IntegratedAreaFilter.java]: @kerner1000 Is it easier to understand? The flag is used to virtually invert the deletion of the peaks.

**Comment by kerner1000** [comment]
See https://github.com/alexstark4/chemclipse/pull/1 for some suggestions.

**Review by kerner1000 — APPROVED** [review]

**Comment by laeubi** [comment]
Please reopen with suitable ECA

**Pull request closed without merging**